src/common/components/repository-row/index.js

```diff
@@ -1,37 +1,75 @@
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import { Link } from 'react-router';
 
-import { Row, Data } from '../vanilla/table-interactive';
+import { Row, Data, Dropdown } from '../vanilla/table-interactive';
 import BuildStatus from '../build-status';
 
 import { parseGitHubRepoUrl } from '../../helpers/github-url';
 
-const RepositoryRow = (props) => {
-  const { snap, latestBuild } = props;
-  const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
-
-  return (
-    <Row>
-      <Data col="30"><Link to={ `/${fullName}/builds` }>{ fullName }</Link></Data>
-      <Data col="20"> </Data>
-      <Data col="20"> </Data>
-      <Data col="30">
-        {/*
-          TODO: show 'Loading' when waiting for status?
-            and also show 'Never built' when no builds available
-        */}
-        { latestBuild &&
-          <BuildStatus
-            link={ `/${fullName}/builds/${latestBuild.buildId}`}
-            status={ latestBuild.status }
-            statusMessage={ latestBuild.statusMessage }
-            dateStarted={ latestBuild.dateStarted }
-          />
-        }
-      </Data>
-    </Row>
-  );
-};
+class RepositoryRow extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      notConfiguredDropdownExpanded: false
+    };
+  }
+
+  onConfiguredClick() {
+    this.setState({
+      notConfiguredDropdownExpanded: !this.state.notConfiguredDropdownExpanded
+    });
+  }
+
+  renderNotConfiguredDropdown() {
+    return (
+      <Dropdown>
+        <Row>
+          <Data col="100">
+            This repo needs a snapcraft.yaml file, so that Snapcraft can make it buildable, installable, and runnable.
+            {/* TODO: add more info/links as in spec */}
+          </Data>
+        </Row>
+      </Dropdown>
+    );
+  }
+
+  render() {
+    const { snap, latestBuild } = this.props;
+    const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
+
+    const notConfigured = true;
+    const showNotConfiguredDropdown = notConfigured && this.state.notConfiguredDropdownExpanded;
+
+    const isActive = showNotConfiguredDropdown; // TODO (or any other dropdown)
+    return (
```

Comment:
This diff is much bigger than it should be because whole thing in indented more now.